addons/web_editor/static/src/components/html_history_dialog/html_history_dialog.js

```diff
@@ -0,0 +1,115 @@
+import { _t } from "@web/core/l10n/translation";
+const { DateTime } = luxon;
+
+class HtmlHistoryDialog extends Component {
+    static template = "web_editor.HtmlHistoryDialog";
+    static components = { Dialog };
+    static props = {
+        close: Function,
+        restoreRequested: Function,
+        recordIds: Array,
+        versionnedFieldName: String,
+    };
+
+    state = useState({
+        revisionsData: [],
+        revisionContent: null,
+        revisionComparison: null,
+        revisionId: null,
+        asyncPending: true,
+    });
+    setup() {
+        this.size = 'xl';
+        this.title = _t("History");
+        this.orm = useService("orm");
+
+        onMounted(() => this.init());
+    }
+
+    async init() {
+        await this.getRecordData();
+        this.state.asyncPending = false;
+        await this.updateCurrentRevision(
+            this.state.revisionsData[0].id);
+    }
+
+    async getRecordData(limit=100, offset=0) {
+        // limit the number of record and order them by id desc
+        this.state.revisionsData = await this.orm.call(
+            "field.html.history.revision",
+            "search_read",
+            [],
+            {
+                fields: ["id", "create_date", "create_uid"],
+                domain: [
+                    ["id", "in", this.props.recordIds],
+                    ["res_field", "=", this.props.versionnedFieldName]
+                ],
+                offset: offset,
+                limit: limit,
+                order: "id desc"
+            }
+        );
+    }
+
+    async updateCurrentRevision(revisionId) {
+        if (this.state.currentId === revisionId || this.state.asyncPending) {
+            return;
+        }
+        this.state.asyncPending = true;
+        this.env.services.ui.block();
+        this.state.revisionId = revisionId;
+        this.state.revisionComparison = await this.getRevisionComparison(
+            revisionId);
+        this.state.revisionContent = await this.getRevisionContent(revisionId);
+        this.state.asyncPending = false;
+        this.env.services.ui.unblock();
+    }
+
+    getRevisionComparison = memoize(async function getRevisionComparison(revisionId) {
+        const comparison = await this.orm.call(
+            "field.html.history.revision",
+            "get_comparison",
+            [revisionId]
+        );
+        return markup(comparison);
+    }.bind(this));
+
+    getRevisionContent = memoize(async function getRevisionContent(revisionId) {
+        const content = await this.orm.call(
+            "field.html.history.revision",
+            "get_content",
+            [revisionId]
+        );
+        return markup(content);
+    }.bind(this));
+
+    async _onRestoreRevisionClick() {
+        this.state.buttonDisabled = true;
+        await this.restoreRevision(this.state.revisionId);
+        this.props.close();
+    }
+    async restoreRevision(revisionId) {
```

Comment:
I don't see the point of having a separate function that is called only once just to encapsulate only two lines, but I do recognize restoring a revision is an important concept at play here so if you think it really adds something to the table you can keep it.